main.tf

```diff
@@ -48,6 +48,16 @@ data "oci_identity_region_subscriptions" "home_region" {
   }
 }
 
+# wait for X seconds for things to settle down on the OCI side
+# before trying to create the Lacework external integration
+resource "time_sleep" "wait_time" {
+  create_duration = var.wait_time
+  depends_on = [
```

Comment:
I understand what you are wanting to do here but rather than wait an arbitrary amount of time which may need to be tweaked is there something that you can be sure of that when it is created or ready that this will work correctly

Reply:
@djmctavish I honestly don't know if there is a way to check this. I'm simply copying the precedent we set over in the gcp config module https://github.com/lacework/terraform-gcp-config/blob/main/main.tf#L191

Reply:
No I understand, just exploring the options, if it enables you to get this working then it is good to go

Reply:
I'll wait for @iok0 to confirm this is working as expected.

Reply:
I'm wondering if adding `module.lacework_oci_credentials` to the `depends_on` list in the `lacwork_integration` resource will do the trick, i.e. making the module dependency explicit. Have we tried that?

Reply:
@kolbeinn that is already added below. Would you mind syncing up with @iok0 on this one? I believe he has noted some options to resolve this based on his testing

Reply:
@rmoles Sure, I'll sync up with @iok0. To be clear, what I'm suggesting is that we omit the `wait_time` resource, and instead add `module.lacework_oci_credentials` explicitly to the `depends_on` list in `lacework_integration_oci_cfg.lacework_integration`.